src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StackAlloc.cs

```diff
@@ -64,12 +64,20 @@ public override BoundNode VisitStackAllocArrayCreation(BoundStackAllocArrayCreat
                         type: ErrorTypeSymbol.UnknownResultType);
                 }
 
+                // The stackalloc instruction requires that the evaluation stack contains only its parameter when executed.
+                // We arrange to clear the stack by wrapping it in a SpillSequence, which will cause pending computations
+                // to be spilled, and also by storing the result in a temporary local, so that the result does not get
+                // hoisted/spilled into some state machine.  If that temp local needs to be spilled that will result in an
+                // error.
                 _needsSpilling = true;
+                var tempAccess = _factory.StoreToTemp(constructorCall, out BoundAssignmentOperator tempAssignment, syntaxOpt: stackAllocNode.Syntax);
```

Comment:
>StoreToTemp [](start = 42, length = 11)

Should we use `kind: SynthesizedLocalKind.Spill`?

Reply:
No, that would cause things to break again.  Because in DEBUG it would be hoisted to a field, and this `var tmp = stackalloc...` would turn into this `frame.tmp = stackalloc...` which would leave frame on the stack when stackalloc appears in the IL.  That is a violation of the constraint that nothing but the argument to stackalloc may be on the stack where it appears in the IL.

---
In reply to: [315396717](https://github.com/dotnet/roslyn/pull/37711#discussion_r315396717) [](ancestors = 315396717)